DEVELOPING.md

```diff
@@ -132,6 +132,49 @@ sequenceDiagram
 ```
 
 
+### Syft Catalogers
+
+##### Summary
+Catalogers are the way in which syft is able to identify and construct packages given some amount of source metadata. 
+For example, syft can use a `package.json` file in conjunction with a `package-lock.json` file to build a list of javascript packages based on the joining metadata:
+- [Cataloger Construction](https://github.com/anchore/syft/blob/e3d6ffd30e44428b898675922a0474221a7f7dc7/syft/pkg/cataloger/javascript/cataloger.go#L16-L21)
+- [Package Lock Parsing](https://github.com/anchore/syft/blob/e3d6ffd30e44428b898675922a0474221a7f7dc7/syft/pkg/cataloger/javascript/parse_package_lock.go#L43-L90)
```

Comment:
We can probably ignore the `package.json` bit here even though it is used. (Also probably update the links to use `main` instead of a hash.) I'd probably just focus on finding and processing a file, something like:

For example, Syft can locate and process `package-lock.json` files when performing filesystem scans. See: [how to specify file globs](https://github.com/anchore/syft/blob/main/syft/pkg/cataloger/javascript/cataloger.go#L16-L21) 
```go
func NewJavascriptLockCataloger() *generic.Cataloger {
	return generic.NewCataloger("javascript-lock-cataloger").
		WithParserByGlobs(parsePackageLock, "**/package-lock.json")
}
```
... and an implementation of the [package-lock.json parser](https://github.com/anchore/syft/blob/main/syft/pkg/cataloger/javascript/parse_package_lock.go#L43-L90)
```go
func parsePackageLock(resolver source.FileResolver, _ *generic.Environment, reader source.LocationReadCloser) ([]pkg.Package, []artifact.Relationship, error) {
	var pkgs []pkg.Package
	dec := json.NewDecoder(reader)
	...
```

Reply:
Latest commit takes the recommendations and implements the suggestions